accepted/0000-hub-online-synchronization.md

```diff
@@ -0,0 +1,359 @@
+
+The new Hub synchronization would be focused on HUB online deployments to allow scale SUSE Manager infrastructure. The only focus of this solution would be to work in scale and make it easier to use.
+
+
+# Detailed design
+[design]: #detailed-design
+
+The solution focuses on re-using the repo-sync mechanism to synchronize channels from Hub server to peripheral servers. It will be described in several steps that aim to address how it will be used and how we can technically support it.
+
+The main focus is to develop the integration between HUB server and peripherals through API only.
+For the Hub/peripheral communication an new API might be useful to have a clear separation. This is needed when we want to make this API organization independent.
+In the existing API the calling user is only allowed to operate on his own organization any maybe utilize "Vendor Channel" when he is a "SUSE Manager Administrator".
+
+Configuration should be pushed from the HUB to peripherals, to avoid the need of direct management of the peripherals.
+
+For now this solution is focused on software channels only, but can be extended to synchronize more data types.
+
+
+## Define connection between HUB and Peripherals
+
+Each peripheral server can only have 1 Hub server (main). This will avoid dealing with problems like channel label conflicts between multiple main servers.
+Configure a hub server on a peripheral server will block access to a set of menu items like: "Admin" -> "Setup wizard" -> "Organization Credentials" and "Products" and "PAYG Connection" (similar to what we already have for ISSv1).
+We cannot add Hub connection if SCC credentials are defined, they should be mutually excluded.
+
+### Alternative 1:
+We can follow a similar approach to what exists on ISSv1. On the hub side we can define multiple peripheral servers to connect to by providing the FQDN and an authentication token.
+On the peripheral side we also need to define the Hub server FQDN and it would generate an Authentication token for he HUB.
+
+To define this connection the existing API namespace should be re-use (`sync.master` and `sync.slave`), with some adaptations to cope with the new requirements. Example of this is the need to define an authentication method between servers to the connection more secure.
+
+During development phase we should consider if we can join both namespaces in one name `hub`, for example.
+
+
+### Alternative 2:
+We can follow a similar approach to what we do to activate a proxy.
+On the Hub we create an API which is called authenticated and can take additional data like credentials for the peripheral server.
+This API create a system (kubernetes) or change an existing system (podman host exists) to an Uyuni Peripheral Server.
```

Comment:
I would separate it from the current systems list and system profile registration. We have made that for hub/peripheral on 5.0 because we didn't have a better approach.
I would not tie the UB/peripheral configuration on the need to have a system profile on SUSE Manager, that is a detail

Reply:
On top of that, I would follow the approach of ISSv1, and have dedicated UI/API call to configure the relationship and not rely on the peripheral being registered as minion or system profile on the HUB

Reply:
I think we need to discuss it how we want to show a Peripheral or Proxy "server" in the UI. 
From my point of view we need to handle both in the same way, but we should use the time now to find a final solution and not a another workaround or short cut.

Reply:
100% agree with you. After our conversation in the morning, I'm better understanding what was your goal.
The initial configuration have a circular dependency, and to avoid to many manual steps we must think of a way to improve it. Using a similar mechanism and the proxy activation makes sense to me. As you have said we could call an API on the HUB, passing the credentials for the peripheral server and we setup the ISSv3 basic configurations automatically. 
 
ON the management side, I see a few questions that can be managed differently:
- Manage a proxy and peripheral servers. In my opinion, this should done through the existing Systems UI with a new element(s). The logic for server and proxy should be similar, with different entitlements.
- Configure HUB-perpheral authentication: this can be done with a new tab on the system details, or on a dedicated page on Admin menu. A dedicated page would be better, in my opinion.
- Configure HUB-perpheral content synchronization: We should make it on a dedicated page on the admin menu.

We can make a differentiation about the system management, like upgrading the peripheral server/proxy, and the ISSv3 configuration and management. They can exist independently of each other, and use just one or the other.